### PR TITLE
Add join table for visits

### PR DESCRIPTION
**Pull request by zieeco**
## vet clinic: Add join table [visit]
**In this milestone, I did the following**
- Added answers to the query 
- Added the intersecting tables
- Updated the animals table

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -68,3 +68,24 @@ SELECT animals.name FROM animals JOIN owners ON animals.owner_id = owners.id JOI
...
+-- List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.
+SELECT animals.name, visits.date_of_visit FROM animals JOIN visits ON visits.vets_id = animals.id JOIN vets ON vets.id = visits.vets_id WHERE vets.name = 'Stephanie Mendez' AND visits.date_of_visit >= '2020-04-01' AND visits.date_of_visit <= '2020-08-30';
+-- What animal has the most visits to vets?
+SELECT animals.name As animal_name, COUNT(*) As total_visits FROM animals JOIN visits ON visits.animals_id = animals.id GROUP BY animals.name ORDER BY COUNT(*) DESC LIMIT 1;
+-- Who was Maisy Smith's first visit?
+SELECT vets.name As vets_name, animals.name animal_name, date_of_visit FROM animals JOIN visits ON visits.animals_id = animals.id JOIN vets ON vets.id = visits.vets_id WHERE vets.name = 'Maisy Smith' ORDER BY date_of_visit LIMIT 1;
+-- Details for most recent visit: animal information, vet information, and date of visit.
+SELECT a.name AS ani_name, vt.name AS vet_name, date_of_visit FROM visits v JOIN animals a ON v.animals_id = a.id JOIN vets vt ON v.vets_id = vt.id ORDER BY date_of_visit DESC LIMIT 1;
```
Comment (wandji20) [inline queries.sql:79]: - [ ] Kindly include all details of animals and vets in your query and not just the name

File: queries.sql [hunk 2]
```diff
@@ -68,3 +68,24 @@ SELECT animals.name FROM animals JOIN owners ON animals.owner_id = owners.id JOI
...
+-- How many visits were with a vet that did not specialize in that animal's species?
+SELECT COUNT(*) FROM visits JOIN vets ON vets.id = visits.vets_id JOIN animals ON animals.id = visits.animals_id WHERE vets.id NOT IN (SELECT vets_id FROM specializations WHERE species_id = animals.species_id);
+-- What specialty should Maisy Smith consider getting? Look for the species she gets the most.
+SELECT COUNT(*) AS species FROM visits v FULL OUTER JOIN vets vt ON v.vets_id = vt.id FULL OUTER JOIN specializations s ON s.vets_id = vt.id WHERE species_id IS NULL;
```
Comment (wandji20) [inline queries.sql:81]: - [ ] We need the name of the specie she gets the most.

**Review by wandji20 — CHANGES_REQUESTED** [review]
Hello, @zieeco  :raised_hands: 
-----
Great :muscle:  job :raised_hands:  implementing most of the requirements for this project.
However, there are a few issues you need to address to have this project approved.
## To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Shema :100: 
:heavy_check_mark: Data :100: 

## Status: Require Changes :recycle: :red_circle: 
Please check the comments below and under the review and make all required changes.
Optional comments are not required to mark this project approved

## Optional Changes
- N/A

**Happy coding :computer:** 
**_Do not hesitate to make comments in this PR in case of doubts._**
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Dipeshtwis — APPROVED** [review]
Hi @zieeco, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**